File: src/Services/TravelService/Models/TravelPlan.cs
using System;
using System.Collections.Generic;

namespace TravelService.Models
{
    public class TravelPlan
    {
        public Guid Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public decimal Budget { get; set; }
        public string Notes { get; set; } = string.Empty;
        public Guid UserId { get; set; } // Owner
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        // Navigation properties
        public List<Destination> Destinations { get; set; } = new List<Destination>();
        public List<PackingListItem> PackingListItems { get; set; } = new List<PackingListItem>();
        public List<TravelPlanShare> Shares { get; set; } = new List<TravelPlanShare>();
    }
}

File: src/Frontend/src/models/TravelPlan.js
import { Destination } from './Destination';
import { PackingItem } from './PackingItem';

export class TravelPlan {
  constructor(data = {}) {
    this.id = data.id || '';
    this.name = data.name || '';
    this.description = data.description || '';
    this.startDate = data.startDate ? new Date(data.startDate) : null;
    this.endDate = data.endDate ? new Date(data.endDate) : null;
    this.budget = data.budget || 0;
    this.notes = data.notes || '';
    this.userId = data.userId || '';
    this.createdAt = data.createdAt ? new Date(data.createdAt) : null;
    this.destinations = Array.isArray(data.destinations) 
      ? data.destinations.map(d => new Destination(d)) 
      : [];
    this.packingListItems = Array.isArray(data.packingListItems) 
      ? data.packingListItems.map(i => new PackingItem(i)) 
      : [];
    this.shares = data.shares || [];
  }

  getDaysCount() {
    if (!this.startDate || !this.endDate) return 0;
    const diffTime = Math.abs(this.endDate - this.startDate);
    const diffDays = Math.ceil(diffTime / (1000 * 60 * 60 * 24)) + 1;
    return diffDays;
  }
}
